// capstone-client/src/user/components/Login/Login.jsx
import React, { useState, useEffect } from 'react';
import { Link, useNavigate } from 'react-router-dom';
import './Login.css';

function Login() {
  const [formData, setFormData] = useState({
    email: '',
    password: '',
    rememberMe: false
  });
  const [isLoading, setIsLoading] = useState(false);
  const [error, setError] = useState('');
  const [showTermsModal, setShowTermsModal] = useState(false);
  const [showPrivacyModal, setShowPrivacyModal] = useState(false);
  const navigate = useNavigate();

  useEffect(() => {
    // Check if user is already logged in
    const token = localStorage.getItem('token');
    const isLoggedIn = localStorage.getItem('isLoggedIn') === 'true';
    if (token && isLoggedIn) {
      navigate('/home');
    }
  }, [navigate]);

  // Close modals on escape key
  useEffect(() => {
    const handleEscape = (e) => {
      if (e.key === 'Escape') {
        setShowTermsModal(false);
        setShowPrivacyModal(false);
      }
    };

    document.addEventListener('keydown', handleEscape);
    return () => document.removeEventListener('keydown', handleEscape);
  }, []);

  // Manage body scroll when modals are open
  useEffect(() => {
    if (showTermsModal || showPrivacyModal) {
      document.body.classList.add('modal-open');
    } else {
      document.body.classList.remove('modal-open');
    }

    return () => {
      document.body.classList.remove('modal-open');
    };
  }, [showTermsModal, showPrivacyModal]);

  const handleInputChange = (e) => {
    const { name, value, type, checked } = e.target;
    setFormData(prev => ({
      ...prev,
      [name]: type === 'checkbox' ? checked : value
    }));
    // Clear error when user starts typing
    if (error) setError('');
  };

  const handleSubmit = async (e) => {
    e.preventDefault();
    setIsLoading(true);
    setError('');

    try {
      const response = await fetch('http://localhost:5000/api/login', {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
        },
        body: JSON.stringify({
          email: formData.email,
          password: formData.password
        }),
      });

      const data = await response.json();

      if (response.ok) {
        // Store token and user data
        localStorage.setItem('token', data.token);
        localStorage.setItem('user', JSON.stringify(data.user));
        localStorage.setItem('isLoggedIn', 'true'); // Add this for ProtectedRoute
        
        if (formData.rememberMe) {
          localStorage.setItem('rememberUser', 'true');
        }
        
        // Navigate to home page (not dashboard)
        navigate('/home');
      } else {
        setError(data.message || 'Invalid email or password');
      }
    } catch (error) {
      console.error('Login error:', error);
      setError('Network error. Please try again.');
    } finally {
      setIsLoading(false);
    }
  };

  return (
    <div className="login-container">
      <div className="login-wrapper">
        {/* Header */}
        <div className="login-header">
          <Link to="/" className="back-link">
            <svg width="16" height="16" viewBox="0 0 24 24" fill="currentColor">
              <path d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.42-1.41L7.83 13H20v-2z"/>
            </svg>
            Back to Home
          </Link>
          <div className="brand">
            <img src="/barangay_logo.png" alt="Barangay Logo" className="brand-logo" />
            <div className="brand-text">
              <h1>Welcome Back</h1>
              <p>Sign in to your job portal account</p>
            </div>
          </div>
        </div>

        {/* Login Form */}
        <div className="login-card">
          <form onSubmit={handleSubmit} className={`login-form ${isLoading ? 'loading' : ''}`}>
            {error && (
              <div className="error-alert">
                <svg width="20" height="20" viewBox="0 0 24 24" fill="currentColor">
                  <path d="M12 2C6.486 2 2 6.486 2 12s4.486 10 10 10 10-4.486 10-10S17.514 2 12 2zm4.207 12.793l-1.414 1.414L12 13.414l-2.793 2.793-1.414-1.414L10.586 12 7.793 9.207l1.414-1.414L12 10.586l2.793-2.793 1.414 1.414L13.414 12l2.793 2.793z"/>
                </svg>
                {error}
              </div>
            )}

            <div className="form-group floating">
              <input
                type="email"
                id="email"
                name="email"
                className={`form-input ${formData.email ? 'success' : ''}`}
                placeholder=" "
                value={formData.email}
                onChange={handleInputChange}
                required
                disabled={isLoading}
              />
              <label htmlFor="email" className="form-label">
                Email Address
              </label>
            </div>

            <div className="form-group floating">
              <input
                type="password"
                id="password"
                name="password"
                className={`form-input ${formData.password ? 'success' : ''}`}
                placeholder=" "
                value={formData.password}
                onChange={handleInputChange}
                required
                disabled={isLoading}
              />
              <label htmlFor="password" className="form-label">
                Password
              </label>
            </div>

            <div className="form-options">
              <label className="checkbox-label">
                <input 
                  type="checkbox" 
                  name="rememberMe"
                  checked={formData.rememberMe}
                  onChange={handleInputChange}
                  className="checkbox" 
                />
                <span className="checkbox-text">Remember me</span>
              </label>
              <Link to="/user/forgot-password" className="forgot-link">
                Forgot password?
              </Link>
            </div>

            <button
              type="submit"
              className={`submit-btn ${isLoading ? 'loading' : ''}`}
              disabled={isLoading}
            >
              {isLoading ? (
                <>
                  <div className="spinner"></div>
                  <span>Signing in...</span>
                </>
              ) : (
                'Sign In'
              )}
            </button>
          </form>

          <div className="divider">
            <span>or</span>
          </div>

          <div className="alternative-actions">
            <p className="signup-prompt">
              Don't have an account?{' '}
              <Link to="/user/register" className="signup-link">
                Create account
              </Link>
            </p>
          </div>
        </div>

        {/* Footer */}
        <div className="login-footer">
          <p>
            By signing in, you agree to our{' '}
            <button 
              className="footer-link-btn"
              onClick={() => setShowTermsModal(true)}
            >
              Terms of Service
            </button>
            {' '}and{' '}
            <button 
              className="footer-link-btn"
              onClick={() => setShowPrivacyModal(true)}
            >
              Privacy Policy
            </button>
          </p>
        </div>
      </div>

      {/* Terms of Service Modal */}
      {showTermsModal && (
        <div className="modal-overlay" onClick={() => setShowTermsModal(false)}>
          <div className="modal-container" onClick={(e) => e.stopPropagation()}>
            <div className="modal-header">
              <h2>Terms of Service</h2>
              <button 
                className="modal-close-btn"
                onClick={() => setShowTermsModal(false)}
              >
                <svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
                  <path d="M18 6L6 18M6 6L18 18" stroke="currentColor" strokeWidth="2" strokeLinecap="round" strokeLinejoin="round"/>
                </svg>
              </button>
            </div>
            <div className="modal-content">
              <div className="terms-content">
                <section>
                  <h3>1. Acceptance of Terms</h3>
                  <p>
                    By accessing and using the Barangay Mangan-vaca Job Portal, you accept and agree to be bound by the terms and provision of this agreement.
                  </p>
                </section>

                <section>
                  <h3>2. Use License</h3>
                  <p>
                    Permission is granted to temporarily use the Barangay Mangan-vaca Job Portal for personal, non-commercial transitory viewing only. This is the grant of a license, not a transfer of title, and under this license you may not:
                  </p>
                  <ul>
                    <li>Modify or copy the materials</li>
                    <li>Use the materials for any commercial purpose or for any public display</li>
                    <li>Attempt to reverse engineer any software contained on the portal</li>
                    <li>Remove any copyright or other proprietary notations from the materials</li>
                  </ul>
                </section>

                <section>
                  <h3>3. User Responsibilities</h3>
                  <p>Users of the job portal agree to:</p>
                  <ul>
                    <li>Provide accurate and truthful information</li>
                    <li>Maintain the confidentiality of their account credentials</li>
                    <li>Use the portal in a lawful and respectful manner</li>
                    <li>Report any inappropriate content or behavior</li>
                  </ul>
                </section>

                <section>
                  <h3>4. Job Postings and Applications</h3>
                  <p>
                    Employers are responsible for the accuracy of job postings. Job seekers are responsible for the accuracy of their application materials. The Barangay does not guarantee employment outcomes.
                  </p>
                </section>

                <section>
                  <h3>5. Privacy and Data Protection</h3>
                  <p>
                    Personal information collected through this portal will be used solely for employment matching purposes and community development. Please refer to our Privacy Policy for detailed information.
                  </p>
                </section>

                <section>
                  <h3>6. Limitation of Liability</h3>
                  <p>
                    Barangay Mangan-vaca shall not be liable for any damages arising from the use of this portal or any employment arrangements made through it.
                  </p>
                </section>

                <section>
                  <h3>7. Contact Information</h3>
                  <p>
                    For questions about these Terms of Service, please contact the Barangay Mangan-vaca office during regular business hours.
                  </p>
                </section>
              </div>
            </div>
          </div>
        </div>
      )}

      {/* Privacy Policy Modal */}
      {showPrivacyModal && (
        <div className="modal-overlay" onClick={() => setShowPrivacyModal(false)}>
          <div className="modal-container" onClick={(e) => e.stopPropagation()}>
            <div className="modal-header">
              <h2>Privacy Policy</h2>
              <button 
                className="modal-close-btn"
                onClick={() => setShowPrivacyModal(false)}
              >
                <svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
                  <path d="M18 6L6 18M6 6L18 18" stroke="currentColor" strokeWidth="2" strokeLinecap="round" strokeLinejoin="round"/>
                </svg>
              </button>
            </div>
            <div className="modal-content">
              <div className="privacy-content">
                <section>
                  <h3>Information We Collect</h3>
                  <p>
                    The Barangay Mangan-vaca Job Portal collects the following types of information:
                  </p>
                  <ul>
                    <li><strong>Personal Information:</strong> Name, email address, phone number, and address</li>
                    <li><strong>Professional Information:</strong> Work experience, education, skills, and employment preferences</li>
                    <li><strong>Usage Data:</strong> How you interact with our portal, including pages visited and features used</li>
                  </ul>
                </section>

                <section>
                  <h3>How We Use Your Information</h3>
                  <p>We use the collected information for:</p>
                  <ul>
                    <li>Facilitating job matching between employers and job seekers</li>
                    <li>Improving our services and user experience</li>
                    <li>Communicating important updates about the portal</li>
                    <li>Ensuring the security and integrity of our platform</li>
                    <li>Generating anonymized reports for community development</li>
                  </ul>
                </section>

                <section>
                  <h3>Information Sharing</h3>
                  <p>
                    We do not sell, trade, or otherwise transfer your personal information to third parties without your consent, except:
                  </p>
                  <ul>
                    <li>When you apply for a job, relevant information is shared with the employer</li>
                    <li>When required by law or to protect our rights</li>
                    <li>With trusted service providers who assist in operating our portal</li>
                  </ul>
                </section>

                <section>
                  <h3>Data Security</h3>
                  <p>
                    We implement appropriate security measures to protect your personal information against unauthorized access, alteration, disclosure, or destruction. However, no method of transmission over the internet is 100% secure.
                  </p>
                </section>

                <section>
                  <h3>Your Rights</h3>
                  <p>You have the right to:</p>
                  <ul>
                    <li>Access and update your personal information</li>
                    <li>Request deletion of your account and data</li>
                    <li>Opt-out of non-essential communications</li>
                    <li>Request a copy of your data</li>
                  </ul>
                </section>

                <section>
                  <h3>Cookies and Tracking</h3>
                  <p>
                    Our portal uses cookies to enhance user experience, remember preferences, and analyze usage patterns. You can control cookie settings through your browser.
                  </p>
                </section>

                <section>
                  <h3>Contact Us</h3>
                  <p>
                    If you have questions about this Privacy Policy or how we handle your data, please contact the Barangay Mangan-vaca office or email us through the portal's contact form.
                  </p>
                </section>

                <section>
                  <h3>Policy Updates</h3>
                  <p>
                    This Privacy Policy may be updated periodically. We will notify users of significant changes through the portal or via email.
                  </p>
                </section>
              </div>
            </div>
          </div>
        </div>
      )}
    </div>
  );
}

export default Login;
